app/models/manager_refresh/inventory_collection.rb

```diff
@@ -124,6 +124,10 @@ def <<(inventory_object)
       end
     end
 
+    def build(hash)
+      self.<<(new_inventory_object(hash))
```

Comment:
maybe it would be nicer to just call:

```ruby
self << new_inventory_object(hash)
```

? Or that doesn't work for some reason?